pkg/kernel/kernel.go

```diff
@@ -120,24 +132,27 @@ func (k *kernel) Run() {
 	// do not allow config changes anymore
 	k.started.Poison()
 
-	defer k.logger.Info("leaving kernel")
+	defer k.exit()
```

Comment:
Should we document that this function never returns normally (unless configured otherwise)?

Reply:
do you mean comment in the code or some explanation in the kernel docs?

Reply:
Comment in the code

Reply:
added